File: backend/models/resource.py
from typing import Literal

from pydantic import BaseModel

from backend.models.character import Quality

ResourceCategory = Literal["Currency", "Gift", "Item", "Material", "Summoning", "Shard"]


class Resource(BaseModel):
    name: str
    description: str
    category: ResourceCategory
    quality: Quality
    last_updated: int | None = None

File: frontend/src/types/resource.ts
import type { Quality } from './character';

export type ResourceCategory =
  | 'Currency'
  | 'Gift'
  | 'Item'
  | 'Material'
  | 'Summoning'
  | 'Shard';

export interface Resource {
  id?: number;
  name: string;
  description: string;
  category: ResourceCategory;
  quality: Quality;
  last_updated: number;
}
